### DISCUSSION TITLE
Jpec decoder question : TJpg_Decoder

### DISCUSSION THREAD

**Discussion by b1gmans** [Bodmer/TFT_eSPI#2445] [General]
Hi, apologies for posting in the TFT_eSPI discussion, but I could not see a discussion tab under your [TJpg_Decoder](https://github.com/Bodmer/TJpg_Decoder). I am using both libraries and work perfectly - thank you.

My question is a more generic requirement.  I plan on making a little unit that will show family photos - from sd card and tft screen - you know the sort of thing. To make it more interesting, it also will become a clock:  I plan to edit a bunch of family photos and add a black rectangle somewhere in each image that is appropriate to the image  - then, as Im displaying that image to the tft, I want to display the current time in that black square, so i need to know where it is in any given image as I decode the image to display it.   So my question is: How can I figure out the x and y coordinates of the top left corner of my black rectangle? - lets say my images are 480X320 and somewhere in there I have a black rectangle 60X30 pixels - how can I find it as the jpg is decoded?   I could insert another image rather than a black rectangle, if this would make it easier to locate?
Any pointers on how to do this, would be greatly appreciated.

**Comment by Bodmer** [comment]
One way is to number the photos, 0.jpg, 1.jpg etc and create x, y  position arrays in the sketch that define where each time block should be drawn. So for n.jpg the position would be x[n]. y[n].
